README.md

```diff
@@ -0,0 +1,167 @@
+
+`let messages = [];`
+
+Note, that in the line above we are using the new ES6 variable assignment keyword `let`. You will need to put the compiler into strict mode for this to work, otherwise you'll likely get an error when you run the code.
+
+Next let's define a new route for submitting messages. In order to send data to the server, we want to use a different HTTP request "method." For the Hello World server we used a standard GET request, but for sending messages we'll use the POST method. A code stub like this will get us started:
+
+```javascript
+route.post('/message', (request, response) => {
+  // Save the message and send the message id back to the client.
+});
+```
+
+At this point, our server can catch a POST request to the path `/message`. Great! But how will get our message data into the system? Now, normally you'd probably have a web frontend that submits the data through the browser. As mentioned above, we'll simulate this by using curl. The following command will submit a message to our new path:
+
+```bash
+curl -X POST http://localhost:3000/message -d '{"id":1,"message":"Our first message."}'
+```
+
+Okay, so this is how we'll submit data, but how will we read in our route's logic? Here's where we will use the `request` object that is passed to ur callback function as an argument. However, although this object contains a lot of information about the request being made to the server, by default it does not contain the data that we've sent across in our curl request in a way that is easily accessed. To make things a bit easier, we'll use another Express module called [body-parser](https://github.com/expressjs/body-parser). As it's name suggests, this module parses data that the server receives from the request. It then attaches this parsed data to the request object's `body` parameter.
+
+The body-parser is also a special kind of module known as "middleware." You'll be using middleware frequently when proramming in Node. As its name suggests, middleware software that runs in between; that is, it is code that is executed after a request is sent to the server, and before the request is processed by the server itself. 
+
+![Diagram of middlware]()
+
+In order to run the body-parser middleware, we need to register it with our router, like so:
+
+```javascript
+ // Import the body-parser. Place this line near your other imports
+ // at the top of your server.js file.
+ const bodyParser = require('body-parser');
+
+ // Now register the middleware with our router, indicating that we
+ // want to use the parser for JSON data.
+ router.use(bodyParser.json());
+```
+
+Now that we've got the body-parser in place, we are finally in a position to submit our message to the server. Excited? Good, I hope so. Let's see if you can get those tests to pass by yourself. To get you'll started here are a few hints and things to consider:
```

Comment:
"To get you'll started..." -> "Here are a few..." (I would just cut the first bit)